RecoEgamma/EgammaElectronProducers/plugins/ElectronSeedProducer.cc

```diff
@@ -230,16 +230,13 @@ void ElectronSeedProducer::produce(edm::Event& e, const edm::EventSetup& iSetup)
   }
 
   // store the accumulated result
-  std::unique_ptr<ElectronSeedCollection> pSeeds(seeds);
-  ElectronSeedCollection::iterator is;
-  for (is = pSeeds->begin(); is != pSeeds->end(); is++) {
-    edm::RefToBase<CaloCluster> caloCluster = is->caloCluster();
-    SuperClusterRef superCluster = caloCluster.castTo<SuperClusterRef>();
-    LogDebug("ElectronSeedProducer") << "new seed with " << (*is).nHits() << " hits"
-                                     << ", charge " << (*is).getCharge() << " and cluster energy "
+  for (auto const& seed : *seeds) {
+    SuperClusterRef superCluster = seed.caloCluster().castTo<SuperClusterRef>();
+    LogDebug("ElectronSeedProducer") << "new seed with " << seed.nHits() << " hits"
```

Comment:
should the loop be placed in ifdef ```EDM_ML_DEBUG``` ? ... that's unless it's obvious that the compiler ignores this loop otherwise.

Reply:
I think it won't be ignored, I will do as you suggest.